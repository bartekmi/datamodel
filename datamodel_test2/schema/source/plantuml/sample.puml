@startuml

' Simple abstract base class
class FileSystemObject {
  Integer Id
  String Name
}

' Directory inherits from FileSystemObject
class Directory
Directory --|> FileSystemObject

' File inherits from FileSystemObject
class File {
  Bytes Data
}
File --|> FileSystemObject

' SymLink inherits from FileSystemObject and references another FileSystemObject
class SymLink {
  String Path
}
SymLink --|> FileSystemObject
SymLink "0..*" --> "1" FileSystemObject : target

' A Directory contains many FileSystemObjects
Directory "1" --> "0..*" FileSystemObject : contents

' Process class (new)
class Process {
  Integer Pid
  String State
}

' Thread class (new)
class Thread {
  Integer Id
  String Name
}

' Process has many threads (aggregation)
Process "1" o-- "0..*" Thread : threads

@enduml
