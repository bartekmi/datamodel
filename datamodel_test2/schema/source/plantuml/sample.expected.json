{
  Title: PlantUML Model,
  Models: [
    {
      Name: FileSystemObject,
      QualifiedName: FileSystemObject,
      AllProperties: [
        {
          DataType: Integer,
          Name: Id
        },
        {
          DataType: String,
          Name: Name
        }
      ]
    },
    {
      SuperClassName: FileSystemObject,
      Name: Directory,
      QualifiedName: Directory,
      AllProperties: [
        {
          DataType: ID,
          Name: contents
        }
      ]
    },
    {
      SuperClassName: FileSystemObject,
      Name: File,
      QualifiedName: File,
      AllProperties: [
        {
          DataType: Bytes,
          Name: Data
        }
      ]
    },
    {
      SuperClassName: FileSystemObject,
      Name: SymLink,
      QualifiedName: SymLink,
      AllProperties: [
        {
          DataType: String,
          Name: Path
        },
        {
          DataType: ID,
          Name: target
        }
      ]
    },
    {
      Name: Process,
      QualifiedName: Process,
      AllProperties: [
        {
          DataType: Integer,
          Name: Pid
        },
        {
          DataType: String,
          Name: State
        },
        {
          DataType: ID,
          Name: threads
        }
      ]
    },
    {
      Name: Thread,
      QualifiedName: Thread,
      AllProperties: [
        {
          DataType: Integer,
          Name: Id
        },
        {
          DataType: String,
          Name: Name
        }
      ]
    }
  ],
  Associations: [
    {
      OwnerSide: SymLink,
      OwnerMultiplicity: ZeroOrOne,
      OtherSide: FileSystemObject,
      OtherRole: target,
      OtherMultiplicity: Many
    },
    {
      OwnerSide: Directory,
      OwnerMultiplicity: One,
      OtherSide: FileSystemObject,
      OtherRole: contents,
      OtherMultiplicity: Many
    },
    {
      OwnerSide: Process,
      OwnerMultiplicity: Aggregation,
      OtherSide: Thread,
      OtherRole: threads,
      OtherMultiplicity: Many
    }
  ],
  Interfaces: {}
}